src/stcal/jump/twopoint_difference.py

```diff
@@ -152,186 +157,203 @@ def find_crs(
+                    masked_ratio = np.ma.masked_greater(ratio, normal_rej_thresh)
+                    #  The jump mask is the ratio greater than the threshold and the difference is usable
+                    jump_mask = np.logical_and(masked_ratio.mask, np.logical_not(first_diffs_masked.mask))
+                    gdq[:, 1:, :, :] = np.bitwise_or(gdq[:, 1:, :, :], jump_mask *
+                                                     np.uint8(dqflags["JUMP_DET"]))
+                else:  # low number of diffs requires iterative flagging
+                    # calculate the differences between adjacent groups (first diffs)
+                    # use mask on data, so the results will have sat/donotuse groups masked
+                    first_diffs = np.abs(np.diff(dat, axis=1))
+
+                    # calc. the median of first_diffs for each pixel along the group axis
+                    median_diffs = calc_med_first_diffs(first_diffs)
+
+                    # calculate sigma for each pixel
+                    sigma = np.sqrt(np.abs(median_diffs) + read_noise_2 / nframes)
+                    # reset sigma so pxels with 0 readnoise are not flagged as jumps
+                    sigma[np.where(sigma == 0.0)] = np.nan
+
+                    # compute 'ratio' for each group. this is the value that will be
+                    # compared to 'threshold' to classify jumps. subtract the median of
+                    # first_diffs from first_diffs, take the abs. value and divide by sigma.
+                    e_jump = first_diffs - median_diffs[np.newaxis, :, :]
+                    ratio = np.abs(e_jump) / sigma[np.newaxis, :, :]
+
+                    # create a 2d array containing the value of the largest 'ratio' for each pixel
+                    warnings.filterwarnings("ignore", ".*All-NaN slice encountered.*", RuntimeWarning)
+                    max_ratio = np.nanmax(ratio, axis=1)
+                    warnings.resetwarnings()
+                    # now see if the largest ratio of all groups for each pixel exceeds the threshold.
+                    # there are different threshold for 4+, 3, and 2 usable groups
+                    num_unusable_groups = np.sum(np.isnan(first_diffs), axis=(0, 1))
+                    int4cr, row4cr, col4cr = np.where(
+                        np.logical_and(ndiffs - num_unusable_groups >= 4, max_ratio > normal_rej_thresh)
+                    )
+                    int3cr, row3cr, col3cr = np.where(
+                        np.logical_and(ndiffs - num_unusable_groups == 3, max_ratio > three_diff_rej_thresh)
+                    )
+                    int2cr, row2cr, col2cr = np.where(
+                        np.logical_and(ndiffs - num_unusable_groups == 2, max_ratio > two_diff_rej_thresh)
+                    )
+
+                    # get the rows, col pairs for all pixels with at least one CR
+#                    all_crs_int = np.concatenate((int4cr, int3cr, int2cr))
```

Comment:
Should remove this line if it's not needed or used.

Reply:
removed